cmake/modules/FindPySpark.cmake

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+#
+# Try to find pyspark
+# Once done this will define
+#
+#  PYSPARK_FOUND - system has pyspark and its dependencies are met
+#
+
+message(STATUS "Looking for pyspark dependency: Java")
+find_package(Java 1.8 COMPONENTS Runtime)
+
+if (Java_Runtime_FOUND)
+    message(STATUS "Found Java ${Java_JAVA_EXECUTABLE}")
+    message(STATUS "Java version ${Java_VERSION_STRING}")
+    message(STATUS "Looking for pyspark dependency: py4j")
+
+    execute_process(
+        COMMAND ${PYTHON_EXECUTABLE_Development_Main} -c "import py4j"
+        RESULT_VARIABLE _PY4J_SEARCH_SUCCESS
+        OUTPUT_VARIABLE _PY4J_VALUES_OUTPUT
+        ERROR_VARIABLE _PY4J_ERROR_VALUE
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    if(${_PY4J_SEARCH_SUCCESS} EQUAL 0)
+        message(STATUS "Package 'py4j' has been succesfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
+        message(STATUS "Import python package: pyspark")
+
+        execute_process(
+            COMMAND ${PYTHON_EXECUTABLE_Development_Main} -c "import pyspark; print(pyspark.__version__)"
+            RESULT_VARIABLE _PYSPARK_SEARCH_SUCCESS
+            OUTPUT_VARIABLE _PYSPARK_VALUES_OUTPUT
+            ERROR_VARIABLE _PYSPARK_ERROR_VALUE
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+
+        string(REGEX MATCH "^[0-9]+\\.[0-9]+\\.[0-9]+" _PYSPARK_VERSION_STRING "${_PYSPARK_VALUES_OUTPUT}")
+
+        if(${_PYSPARK_SEARCH_SUCCESS} EQUAL 0)
+
+            if(${PYTHON_VERSION_STRING_Development_Main} VERSION_LESS "3.6")
+                # Pyspark dropped support for Python < 3.6 in version 3.1.1
+                if(${_PYSPARK_VERSION_STRING} VERSION_GREATER_EQUAL "3.1.1")
+                    message(STATUS "Package 'pyspark' found with version ${_PYSPARK_VERSION_STRING}, which requires Python > 3.5")
+                    message(STATUS "Please install a version of 'pyspark' compatible with your Python executable")
+                    set(PYSPARK_FOUND FALSE)
+                else()
+                    message(STATUS "Package 'pyspark' has been succesfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
```

Comment:
```suggestion
                    message(STATUS "Package 'pyspark' version ${_PYSPARK_VERSION_STRING} has been successfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
```

Reply:
In the next commit the pyspark version string will be automatically handled through `find_package_handle_standard_args` so the output will be more in line with common CMake logic